packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1125,21 +1126,119 @@ class Positioned extends ParentDataWidget<StackRenderObjectWidgetBase> {
+
+  RenderGrid createRenderObject() => new RenderGrid(delegate: delegate);
+
+  void updateRenderObject(RenderGrid renderObject, GridRenderObjectWidgetBase oldWidget) {
+    renderObject.delegate = delegate;
+  }
+}
+
 /// Uses the grid layout algorithm for its children.
 ///
 /// For details about the grid layout algorithm, see [RenderGrid].
-class Grid extends MultiChildRenderObjectWidget {
-  Grid(List<Widget> children, { Key key, this.maxChildExtent })
+class CustomGrid extends GridRenderObjectWidgetBase {
+  CustomGrid(List<Widget> children, { Key key, this.delegate })
     : super(key: key, children: children) {
-    assert(maxChildExtent != null);
+    assert(delegate != null);
+  }
+
+  /// The delegate that controls the layout of the children.
+  final GridDelegate delegate;
+}
+
+/// Uses a grid layout with a fixed column count.
+///
+/// For details about the grid layout algorithm, see [MaxTileWidthGridDelegate].
+class FixedColumnCountGrid extends GridRenderObjectWidgetBase {
+  FixedColumnCountGrid(List<Widget> children, {
+    Key key,
+    this.columnCount,
+    this.tileAspectRatio: 1.0,
+    this.padding: EdgeDims.zero
+  }) : super(key: key, children: children) {
+    assert(columnCount != null);
   }
 
-  final double maxChildExtent;
+  /// The number of columns in the grid.
+  final int columnCount;
 
-  RenderGrid createRenderObject() => new RenderGrid(maxChildExtent: maxChildExtent);
+  /// The ratio of the width to the height of each tile in the grid.
+  final double tileAspectRatio;
+
+  /// The amount of padding to apply to each child.
+  final EdgeDims padding;
+
+  FixedColumnCountGridDelegate get delegate {
+    return new FixedColumnCountGridDelegate(
+      columnCount: columnCount,
+      tileAspectRatio: tileAspectRatio,
+      padding: padding
+    );
+  }
+}
 
-  void updateRenderObject(RenderGrid renderObject, Grid oldWidget) {
-    renderObject.maxChildExtent = maxChildExtent;
+/// Uses a grid layout with a max tile width.
+///
+/// For details about the grid layout algorithm, see [MaxTileWidthGridDelegate].
+class MaxTileWidthGrid extends GridRenderObjectWidgetBase {
+  MaxTileWidthGrid(List<Widget> children, {
+    Key key,
+    this.maxTileWidth,
+    this.tileAspectRatio: 1.0,
+    this.padding: EdgeDims.zero
+  }) : super(key: key, children: children) {
+    assert(maxTileWidth != null);
+  }
+
+  /// The maximum width of a tile in the grid.
+  final double maxTileWidth;
+
+  /// The ratio of the width to the height of each tile in the grid.
+  final double tileAspectRatio;
+
+  /// The amount of padding to apply to each child.
+  final EdgeDims padding;
+
+  MaxTileWidthGridDelegate get delegate {
+    return new MaxTileWidthGridDelegate(
+      maxTileWidth: maxTileWidth,
+      tileAspectRatio: tileAspectRatio,
+      padding: padding
+    );
+  }
+}
+
+/// Supplies per-child data to the grid's [GridDelegate].
+class GridPlacementData extends ParentDataWidget<GridRenderObjectWidgetBase> {
```

Comment:
Only CustomGrid is going to use this, right? I think this should only be allowed for CustomGrid. Non-Custom specific descendants of GridRenderObjectWidgetBase that use the parent data somehow (of which there are none today) should have their own specific ParentDataWidget that is opinionated about the placementData type.

Alternatively, GridPlacementData could itself be generic on two types, the placementData and the target widget. As in:

```dart
class GridPlacementData<T, S> extends ParentDataWidget<T> {
  ...
  final S placementData;
```

(but with better names than T and S)